prov/hook/trace/src/hook_trace.c

```diff
@@ -82,6 +82,25 @@
 			(uint64_t)tag, (uint64_t)ignore, context); \
 	}
 
+#define TRACE_MR_ATTR(ret, dom, len, flags, attr)    \
+	if (!(ret)) { \
+		FI_TRACE(dom->fabric->hprov, FI_LOG_DOMAIN, \
+		         "len %lu, flags 0x%lx\n%s",  (len), (flags), \
+		          fi_tostr((void *)attr, FI_TYPE_MR_ATTR)); \
```

Comment:
fi_tostr() is not threadsafe.  It would be better to use fi_tostr_r() and pass in a buffer.